app/rust/src/parser/structured_msg.rs

```diff
@@ -74,7 +75,14 @@ impl<'a> Domain<'a> {
         let mut buff = [0; 39];
 
         if let Some((key, value)) = self.tuple().iter().nth(display_idx as usize) {
-            out_key[0..key.name().len()].copy_from_slice(key.name());
+            if let Some(m) = out_key.get_mut(0..key.name().len()) {
```

Comment:
same as above, .ok_or(err)? and copy_from_slice should be good, no need to check lengths

Reply:
still iter_mut